rust/azure_iot_operations_mqtt/tests/mock_receive.rs

```diff
@@ -0,0 +1,482 @@
+        .build()
+        .unwrap();
+    let incoming_packets_tx = IncomingPacketsTx::default();
+    let outgoing_packets_rx = OutgoingPacketsRx::default();
+    let mock_server = MockServer::new(incoming_packets_tx.clone(), outgoing_packets_rx.clone());
+    let options = SessionOptionsBuilder::default()
+        .connection_settings(connection_settings)
+        .injected_packet_channels(Some(InjectedPacketChannels {
+            incoming_packets_tx: incoming_packets_tx,
+            outgoing_packets_rx: outgoing_packets_rx,
+        }))
+        .build()
+        .unwrap();
+    let session = Session::new(options).unwrap();
+    (session, mock_server)
+}
+
+/// Common test logic for filtered/unfiltered single receiver tests at QoS 0.
+/// Tests that it can:
+/// - receive messages via recv()
+/// - receive messages via recv_manual_ack()
+/// - no PUBACKS are sent because QoS 0
+/// - no mechanism for sending PUBACKs is exposed
+async fn qos0_single_reciever_test_logic(
+    mock_server: MockServer,
+    mut receiver: SessionPubReceiver,
+    topic_name: &str,
+) {
+    ///////////// Using recv() /////////////
+
+    // Send publish from mock server and receive it via recv()
+    let proto_publish = mqtt_proto::Publish {
+        topic_name: mqtt_proto::topic(topic_name),
+        packet_identifier_dup_qos: mqtt_proto::PacketIdentifierDupQoS::AtMostOnce,
+        retain: false,
+        payload: bytes::Bytes::from("Publish 1"),
+        other_properties: mqtt_proto::PublishOtherProperties::default(),
+    };
+    let expected_publish = proto_publish.clone().into();
+    mock_server.send_publish(proto_publish);
+    assert_eq!(receiver.recv().await.unwrap(), expected_publish);
+
+    // No ack expected for QoS 0
+    mock_server.expect_no_packet();
+
+    ///////////// Using recv_manual_ack() /////////////
+
+    // Send publish from mock server and receive it via recv_manual_ack()
+    let proto_publish = mqtt_proto::Publish {
+        topic_name: mqtt_proto::topic(topic_name),
+        packet_identifier_dup_qos: mqtt_proto::PacketIdentifierDupQoS::AtMostOnce,
+        retain: false,
+        payload: bytes::Bytes::from("Publish 2"),
+        other_properties: mqtt_proto::PublishOtherProperties::default(),
+    };
+    let expected_publish = proto_publish.clone().into();
+    mock_server.send_publish(proto_publish);
+    let response = receiver.recv_manual_ack().await.unwrap();
+    assert_eq!(response.0, expected_publish);
+    assert!(response.1.is_none()); // No manual ack mechanism for QoS 0
+
+    // No ack expected for QoS 0
+    mock_server.expect_no_packet();
+}
+
+#[tokio::test]
+async fn qos0_single_filtered_receiver() {
+    let (session, mock_server) =
+        setup_client_and_mock_server("qos0_single_filtered_receiver_test_client");
+    let managed_client = session.create_managed_client();
+
+    // Start the session run loop
+    tokio::task::spawn(session.run());
+    mock_server.expect_connect_and_accept(true).await;
```

Comment:
for mocks, should we use "clean start true" because we aren't emulating any of the broker behavior for if the session is present?
```suggestion
    mock_server.expect_connect_and_accept(false).await;
```